### PR TITLE
test: enable tests in ci

### PR DESCRIPTION
**Pull request by muchvo**
## Description

1. Enable tests in ci.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-MARL/safety-gymnasium/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/PKU-MARL/safety-gymnasium/blob/main/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: safety_gymnasium/utils/random_generator.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class RandomGenerator:
 
     def __init__(self):
         """Initialize the random number generator."""
-        self.random_generator: np.random.mtrand.RandomState = None
+        self.random_generator: np.random.RandomState = None  # pylint: disable=no-member
```
Comment (XuehaiPan) [inline safety_gymnasium/utils/random_generator.py:61]: `numpy.random.RandomState` is deprecated. Open a new PR to change this to `numpy.random.Generator`. (breaking change, need a revision bump).

Comment (muchvo) [reply safety_gymnasium/utils/random_generator.py]: Got it, thanks a lot

File: tests/test_env.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def test_env(agent_id, env_id, level, render_mode):
     # obs, _ = env.reset(seed=0)
     terminated, truncated = False, False
     ep_ret, ep_cost = 0, 0
-    for step in range(1000):  # pylint: disable=unused-variable
+    for step in range(10):  # pylint: disable=unused-variable
```
Comment (XuehaiPan) [inline tests/test_env.py:37]:
```suggestion
    for _ in range(10):
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          update-environment: true
+
+      - name: Upgrade pip
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+
+      - name: Update .bashrc
+        run: |
+          echo "export MUJOCO_GL=osmesa" >> ~/.bashrc
+          source ~/.bashrc
```
Comment (XuehaiPan) [inline .github/workflows/tests.yml:55]:
```suggestion
      - name: Set MUJOCO_GL
        run: |
          echo "MUJOCO_GL=osmesa" >> "${GITHUB_ENV}"
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+      - name: Set `SETUPTOOLS_ENABLE_FEATURES`
+        if: ${{ matrix.python-version == '3.8' && matrix.os == 'windows-latest' }}
+        run: |
+          Write-Output 'SETUPTOOLS_ENABLE_FEATURES="legacy-editable"' | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+
```
Comment (XuehaiPan) [inline .github/workflows/tests.yml:61]:
```suggestion
```

Only needed for Python 3.7.

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+          update-environment: true
+
+      - name: Upgrade pip
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+
+      - name: Update .bashrc
+        run: |
+          echo "export MUJOCO_GL=osmesa" >> ~/.bashrc
+          source ~/.bashrc
```
Comment (XuehaiPan) [inline .github/workflows/tests.yml:102]:
```suggestion
      - name: Set MUJOCO_GL
        run: |
          echo "MUJOCO_GL=osmesa" >> "${GITHUB_ENV}"
```

**Review by zmsn-2077 — DISMISSED** [review]
After careful review, I agree with these changes.

**Comment by zmsn-2077** [comment]
Need to rebase.

**Review by zmsn-2077 — APPROVED** [review]
LGTM.

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Set MUJOCO_GL
+        run: |
+          echo "MUJOCO_GL=osmesa" >> "${GITHUB_ENV}"
+          echo "PYOPENGL_PLATFORM=osmesa" >> "${GITHUB_ENV}"
+
+      - name: Install Safety-Gymnasium
+        run: |
+          python -m pip install -e .
+          python -m pip install -vvv --editable '.[test]'
```
Comment (XuehaiPan) [inline .github/workflows/tests.yml:66]:
```suggestion
          python -m pip install -vvv --editable '.[test]'
```

File: tests/test_env.py [hunk 7]
```diff
@@ -14,6 +14,7 @@
 # ==============================================================================
 """Test environments."""
 
+
```
Comment (XuehaiPan) [inline tests/test_env.py:17]:
```suggestion
```

**Review by zmsn-2077 — APPROVED** [review]

**Review by Gaiejj — APPROVED** [review]
LGTM.

File: .github/workflows/tests.yml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  test-ubuntu:
+    name: Test for Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (XuehaiPan) [inline .github/workflows/tests.yml:32]: Don't we support Python 3.11?

```suggestion
        python-version: ["3.8", "3.9", "3.10". "3.11"]
```

Comment (zmsn-2077) [reply .github/workflows/tests.yml:32]: In previous tests, Python 3.11 also had some errors.

**Review by XuehaiPan — APPROVED** [review]

**Review by zmsn-2077 — APPROVED** [review]

**Pull request merged**